app/src/main/java/com/gleam/kiwi/net/KiwiClient.kt

```diff
@@ -1,69 +1,80 @@
-    fun getTasks(token: String): Tasks?
+    suspend fun registerTasks(token: String, task: Tasks)
+    suspend fun getTasks(token: String): Tasks?
 }
 
 class KiwiClient(private val kiwiService: KiwiServiceInterFace) :
     KiwiClientInterface {
-    override fun signUp(user: User) {
-        kiwiService.signUp(user).execute()
-    }
+    private var token: String? = null
 
-    override fun revokeUser(user: User) {
-        kiwiService.revokeUser(user).execute()
+    override suspend fun signUp(user: User) {
+        withContext(Dispatchers.IO) {
+            kiwiService.signUp(user).execute()
+        }
     }
 
-    override fun getNewToken(user: User): String? {
-        return kiwiService.getNewToken(user).execute().let {
-            if (it.isSuccessful) {
-                it.body()
+    override suspend fun signIn(user: User) {
+        withContext(Dispatchers.IO) {
+            kiwiService.getNewToken(user).execute().let {
+                if (it.isSuccessful) token = it.body()
             }
-            null
         }
     }
 
-    override fun revokeToken(token: String) {
-        kiwiService.revokeToken(token)
+    override suspend fun revokeUser(user: User) {
+        withContext(Dispatchers.IO) {
+            kiwiService.revokeUser(user).execute()
+        }
     }
 
-    override fun registerTimetable(token: String, timetable: Timetable) {
-        kiwiService.registerTimetable(token, timetable).execute()
+
+    override suspend fun registerTimetable(token: String, timetable: Timetable) {
+        withContext(Dispatchers.IO) {
+            kiwiService.registerTimetable(token, timetable).execute()
+        }
     }
 
-    override fun getTimetable(token: String): Timetable? {
-        return kiwiService.getTimetable(token).execute().let {
-            if (it.isSuccessful) {
-                it.body()
+    override suspend fun getTimetable(token: String): Timetable? {
+        return withContext(Dispatchers.IO) {
+            kiwiService.getTimetable(token).execute().let {
+                if (it.isSuccessful) {
+                    it.body()
+                }
+                null
             }
-            null
         }
     }
 
-    override fun registerTasks(token: String, task: Tasks) {
-        kiwiService.registerTasks(token, task).execute()
+    override suspend fun registerTasks(token: String, task: Tasks) {
+        withContext(Dispatchers.IO) {
+            kiwiService.registerTasks(token, task).execute()
+        }
     }
 
-    override fun getTasks(token: String): Tasks? {
-        return kiwiService.getTasks(token).execute().let {
-            if (it.isSuccessful) {
-                it.body()
+    override suspend fun getTasks(token: String): Tasks? {
+        return withContext(Dispatchers.IO) {
+            kiwiService.getTasks(token).execute().let {
+                if (it.isSuccessful) {
+                    it.body()
+                }
+                null
             }
```

Comment:
Use `takeIf` to check.

```suggestion
            kiwiService.getTasks(token).execute().takeIf { it.isSuccessful() }.let { it.body() }
            }
```

Reply:
これだとisSuccessfulの結果に関係なくit.body()が呼び出されたりしませんか

Reply:
あ、`?`抜けてるからそもそもコンパイル通らなさそう